commands/update.py

```diff
@@ -456,6 +468,164 @@ async def disable_cutoff_tracking(self,
+                        elif attribute == 'cool':
+                            attr_png = "data/img/cool2.png"
+                        elif attribute == 'pure':
+                            attr_png = "data/img/pure2.png"
+                        else:
+                            attr_png = "data/img/happy2.png"
+                        attr_bg = Image.open(attr_png)
+                        im.paste(attr_bg, (132, 2), mask=attr_bg)
+
+                        band_png = f"data/img/band_{band_id}.png"
+                        band_bg = Image.open(band_png)
+                        im.paste(band_bg, (1, 2), mask=band_bg)
+                        # The last URL in the list will always be the trained card which isn't needed for the gacha cards
+                        if card_type in gacha_types:
+                            if not path.exists(f'data/img/icons/{chara_name}/{rarity}/'):
+                                filepath = Path(f'data/img/icons/{chara_name}/{rarity}/')
+                                filepath.mkdir(parents=True, exist_ok=True)
+                            if not path.exists(gacha_icons_path):
+                                if int(rarity) == 2:
+                                    im.save(gacha_icons_path)
+                                elif int(rarity) > 2:
+                                    if card_type == 'birthday':
+                                        im.save(gacha_icons_path)
+                                    elif url != url_list[-1]:
+                                        im.save(gacha_icons_path)
+                        im.save(full_icons_path)
+                    except Exception as e:
+                        print(f"Failed adding card with ID {card_id} ({e})")
+                        pass
+                print(f'updated card {card_id}')
+        except Exception as e:
+            print(f"Failed adding card with ID {card_id} ({e})")
+            pass
+
+    async def update_card_icons(self):
+        card_api = await self.fetch_api("https://bestdori.com/api/cards/all.5.json")
+        chara_api = await self.fetch_api("https://bestdori.com/api/characters/all.2.json")
+        if not path.exists('data/img/icons/base_icons/'):
+            filepath = Path('data/img/icons/base_icons/')
+            filepath.mkdir(parents=True, exist_ok=True)
+        if not path.exists('data/img/icons/full_icons/'):
+            filepath = Path('data/img/icons/full_icons/')
+            filepath.mkdir(parents=True, exist_ok=True)
+        await asyncio.gather(*[self.generate_card_icon(card_id, card_api, chara_api) for card_id in card_api])
+
+    @discord.slash_command(name='updatecards', description='Update card images manually')
+    async def update_cards_command(self, ctx: discord.ApplicationContext):
+        await ctx.interaction.response.defer()
+        await self.update_card_icons()
+        await ctx.interaction.followup.send("Cards successfully updated.")
+
+    @tasks.loop(seconds=300.0)
```

Comment:
Does this really need to be run every 300 seconds? Feels to me that this should run maybe at most once a week, when patches come out.